src/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.StateManager.ProjectStates.cs

```diff
@@ -121,37 +120,22 @@ private async Task<ProjectAnalyzerStateSets> UpdateProjectStateSetsAsync(Project
                 return projectStateSets.Value;
             }
 
-            private ProjectAnalyzerReferenceChangedEventArgs? GetProjectAnalyzerReferenceChangedEventArgs(
-                Project project,
-                ImmutableDictionary<object, ImmutableArray<DiagnosticAnalyzer>> newMapPerReference,
-                ImmutableDictionary<DiagnosticAnalyzer, StateSet> newMap)
+            private ProjectAnalyzerReferenceChangedEventArgs? GetProjectAnalyzerReferenceChangedEventArgs(Project project)
             {
                 // No need to use _projectAnalyzerStateMapGuard during reads of _projectAnalyzerStateMap
                 if (!_projectAnalyzerStateMap.TryGetValue(project.Id, out var entry))
                 {
-                    // no previous references and we still don't have any references
-                    if (newMap.Count == 0)
-                    {
-                        return null;
-                    }
-
                     // new reference added
-                    return new ProjectAnalyzerReferenceChangedEventArgs(project, newMap.Values.ToImmutableArrayOrEmpty(), []);
+                    return null;
```

Comment:
converted to 'null', since hte only code that reads the 'removed' argument bails out if it is empty.